alphastats/gui/pages/05_Analysis.py

```diff
@@ -92,9 +94,13 @@ def show_start_llm_button(analysis_method: str) -> None:
         if StateKeys.LLM_INTEGRATION in st.session_state:
             del st.session_state[StateKeys.LLM_INTEGRATION]
         st.session_state[StateKeys.LLM_INPUT] = (analysis_object, parameters)
+        with st.spinner("Retrieving uniprot data on regulated features ..."):
```

Comment:
This takes quite some time, a quickwin could be to use a progress bar here: https://github.com/Wirg/stqdm

Currently we need it only for LLM part, could we move it there?

Local caching of data could also be easily done (e.g. dump the dict to a json) => ticket?

Reply:
I put it here, because this is when the user decides to move on with an LLM based analysis for the data. Putting it on the LLM page would mean another button click there before actually doing the analysis. But that is just what feels natural to me - happy to change upon more user feedback.

Reply:
Implemented the stqdm, thank you for the suggestion
Added ticket for caching to the project board.